listenbrainz/webserver/views/stats_api.py

```diff
@@ -498,6 +503,35 @@ def get_daily_activity(user_name: str):
     }})
 
 
+@stats_api_bp.route("/user/<user_name>/artist-map")
+@crossdomain()
+@ratelimit()
+def get_artist_map(user_name: str):
+    user = db_user.get_by_mb_id(user_name)
+    if user is None:
+        raise APINotFound("Cannot find user: {}".format(user_name))
+
+    stats_range = request.args.get('range', default='all_time')
+    if not _is_valid_range(stats_range):
+        raise APIBadRequest("Invalid range: {}".format(stats_range))
+
+    stats = db_stats.get_user_artists(user['id'], stats_range)
+    if stats is None or getattr(stats, stats_range) is None:
+        raise APINoContent('')
+
+    artist_msids = [artist.artist_msid for artist in getattr(
+        stats, stats_range).artists if artist.artist_msid is not None]
+    country_code_data = _get_country_codes_by_msids(artist_msids)
```

Comment:
We should consider storing this in the database once it's been calculated (and updating every few days or so). This means that the endpoint will only be slow once.